### ISSUE TITLE
PDF buffer does not revert after AUCTeX compilation when TeX-recenter-output-buffer has been called during compilation

### ISSUE THREAD

**Issue by adamliter** [politza/pdf-tools#499]
I'm not totally sure if this is an AUCTeX bug or a pdf-tools bug. I think it might be a pdf-tools bug, so I'm reporting it here first.

I use AUCTeX, with pdf-tools as my default viewer. I've added

```lisp
(add-hook 'TeX-after-compilation-finished-functions #'TeX-revert-document-buffer)
```

to my configuration so that the PDF buffer reverts after a TeX compilation.

During the compilation process, if I invoke `C-c C-l` (`TeX-recenter-output-buffer`) in order to see the results of the compilation, the PDF buffer does *not* revert.

If I do not invoke `C-c C-l`, the PDF buffer *does* revert.

The reason I think this might be a pdf-tools bug is because this does not happen with DocView. Moreover, on Emacs.SE, someone [really helpfully tried to debug this with me](https://chat.stackexchange.com/rooms/95504/update-pdf-view-after-latex-compilation-https-emacs-stackexchange-com-questions), and `pdf-view-revert-buffer` is successfully run after successful compilation. Still being new to Emacs and Emacs Lisp, I'm not sure what more to do to debug this.

Any idea if this might be a pdf-tools bug? I can also try reporting it to the AUCTeX mailing list, but I figured I would start here since everything seems to work as intended with DocView.

**Comment by politza** [comment]
Is the PDF buffer still displayed, after you've started the compilation and called TeX-recenter-output-buffer ?  Or is it not displayed, but you switched to it after the compilation and noticed that it did not display the changed content ?

**Comment by adamliter** [comment]
So I usually have two different windows open, one for the .tex file and one for the PDF. My point is in the window for the .tex file, then I run `C-c C-a` (`TeX-command-run-all`), then there's a message in the minibuffer that says "Type `C-c C-l` to display results of compilation", so I type `C-c C-l`. This causes the window with the PDF to switch to displaying the buffer for the compilation. So now I have two windows, one with the .tex file and one with the compilation. Once the compilation finishes, the window switches back to displaying the PDF buffer automatically (because I ran `TeX-command-run-all`, which will run the "View" command), but the buffer does not revert; it still displays the old version of the PDF. As soon as I do anything in the buffer, however, it will revert. For example, paging up or down or anything like that causes the buffer to revert. It just doesn't revert until I do so something in the buffer.

If I do all of the sames things but I do *not* press `C-c C-l` prior to the compilation finishing, the PDF buffer *does* revert without me doing anything.

**Comment by politza** [comment]
Ok, thanks.  I pushed a change, please see if that works for you.

**Comment by adamliter** [comment]
@politza Thanks, yes, the issue seems to be fixed by db7de3901ae0e55f6ab8cf9baec257f706c3d16e. After updating from MELPA, it now works as expected.
